test/common/network/resolver_impl_test.cc

```diff
@@ -0,0 +1,126 @@
+#include <cstdint>
+#include <list>
+#include <string>
+
+#include "envoy/common/exception.h"
+#include "envoy/network/resolver.h"
+#include "envoy/registry/registry.h"
+
+#include "common/common/thread.h"
+#include "common/network/address_impl.h"
+#include "common/network/resolver_impl.h"
+
+#include "test/mocks/network/mocks.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Network {
+namespace Address {
+class IpResolverTest : public testing::Test {
+public:
+  ResolverFactory* factory_{Registry::FactoryRegistry<ResolverFactory>::getFactory("envoy.ip")};
+};
+
+TEST_F(IpResolverTest, Basic) {
+  auto address = factory_->create()->resolve("1.2.3.4", 443);
+  EXPECT_EQ(address->ip()->addressAsString(), "1.2.3.4");
+  EXPECT_EQ(address->ip()->port(), 443);
+}
+
+TEST_F(IpResolverTest, DisallowsNamedPort) {
+  auto resolver = factory_->create();
+  EXPECT_THROW(resolver->resolve("1.2.3.4", "http"), EnvoyException);
```

Comment:
Could be good to validate it's the exception we're after with `EXPECT_THROW_WITH_MESSAGE`. Do we have full coverage of all the `throw` cases in the PR?